gdp/small_batch/gdp_small_batch.py

```diff
@@ -9,99 +17,200 @@
+    m.b = pe.Var(m.i, within=pe.NonNegativeReals, doc='Batch size of product i [L]')  # Batch size of product i [L]
+    m.tl = pe.Var(m.i, within=pe.NonNegativeReals, doc='Cycle time of product i [hr]')  # Cycle time of product i [hr]
     # Number of units in parallel stage j
-    m.n = pe.Var(m.j, within=pe.NonNegativeReals)
+    m.n = pe.Var(m.j, within=pe.NonNegativeReals, doc='Number of units in parallel stage j')
 
     # Constraints
 
     # Volume requirement in stage j
     @m.Constraint(m.i, m.j)
     def vol(m, i, j):
+        """
+        Volume Requirement for Stage j.
+
+        Args:
+            m (pyomo.ConcreteModel): small batch GDP model
+            i (str): product
+            j (str): stage
+        
+        Returns:
+            Constraint
+        """
         return m.v[j] >= pe.log(m.s[i, j]) + m.b[i]
 
     # Cycle time for each product i
     @m.Constraint(m.i, m.j)
     def cycle(m, i, j):
+        """
+        Cycle time for each product i.
+
+        Args:
+            m (pyomo.ConcreteModel): small batch GDP model
+            i (str): product
+            j (str): stage
+
+        Returns:
+            Constraint
+        """
         return m.n[j] + m.tl[i] >= pe.log(m.t[i, j])
 
     # Constraint for production time
     @m.Constraint()
     def time(m):
+        """
+        Production time constraint.
+
+        Args:
+            m (pyomo.ConcreteModel): small batch GDP model
+
+        Returns:
+            Constraint
+        """
         return sum(m.q[i]*pe.exp(m.tl[i]-m.b[i]) for i in m.i) <= m.h
 
     # Relating number of units to 0-1 variables
     @m.Constraint(m.j)
     def units(m, j):
+        """
+        Relating number of units to 0-1 variables.
+
+        Args:
+            m (pyomo.ConcreteModel): small batch GDP model
+            j (str): stage
+        
+        Returns:
+            Constraint
+        """
         return m.n[j] == sum(m.coeffval[k, j] for k in m.k)
 
     # Only one choice for parallel units is feasible
     @m.LogicalConstraint(m.j)
     def lim(m, j):
+        """
+        Only one choice for parallel units is feasible.
+
+        Args:
+            m (pyomo.ConcreteModel): small batch GDP model
+            j (str): stage
+        
+        Returns:
+            Logical Constraint
+        """
         return pe.exactly(1, m.Y[1, j], m.Y[2, j], m.Y[3, j])
 
     # _______ Disjunction_________
 
     def build_existence_equations(disjunct, k, j):
+        """
+        Build the Logic Proposition (euqations) for the existence of the stage.
+
+        Args:
+            disjunct (pyomo.gdp.Disjunct): Disjunct block
+            k (int): number of parallel units
+            j (str): stage
+
+        Returns:
+            None, the proposition is built inside the function 
+        """
         m = disjunct.model()
 
         # Coeffval activation
         @disjunct.Constraint()
         def coeffval_act(disjunct):
+            """
+            Coeffval activation.
```

Comment:
Write the equation here

Reply:
Added the Equation.

Reply:
Why did you add log(k) to this? The constraint does not include this